### ISSUE TITLE
Broken visualization of invalid input token in multiline input

### ISSUE THREAD

**Issue by Majkl578** [hoaproject/Compiler#97]
When using multiline input, the arrow indicating position of invalid token is not positioned correctly:
```php
/**
 * @Foo(hello
 */
```
When error is the `(` (i.e. missing pair `)`), the exception message currently contains:
```
/**
 * @Foo(hello
 */
                   ↑
```

What would be expceted instead:
```
/**
 * @Foo(hello
        ↑
 */
```
or:
```
/**
 * @Foo(hello
        ↑
```

**Comment by flip111** [comment]
@Majkl578 i might take a look at this bug, but please make it easy for me and provide runnable test code

**Comment by Majkl578** [comment]
I'll see what can be done, but Hoa's proprietary test suite makes things hard for contributors.

**Comment by flip111** [comment]
@Majkl578 what do you mean? proprietary is usually used to refer to license. In this case it's not proprietary but BSD license.

As for test framework it's based on http://atoum.org/ instead of phpunit as far as i know. I used atoum once it was pretty good.

You can just provide a snippet of code that shows your error, don't need a test case at the moment.

**Comment by flip111** [comment]
It would be good to fix this bug together with https://github.com/hoaproject/Compiler/issues/78 at the same time.

**Comment by Majkl578** [comment]
By _proprietary_ I meant it's an in-house suite (hoa/test), not a standard tool like PHPUnit. So a contributor must first spend time investigating how this tool works and how the tests look like. It's an entry barrier.

**Comment by Majkl578** [comment]
Also, _proprietary_ doesn't only mean _properietary license_, in this sense it should be understood as _non-standard_ ([see Wiktionary, adj. 4](https://en.wiktionary.org/wiki/proprietary)).

**Comment by flip111** [comment]
@Majkl578 you are right PHPUnit is not Atoum. PHPUnit is the most used test framework for PHP, but a standard? Anyway we can worry about unit test later, i can do it if necessary, it's not that difficult
